src/algorithm.rs

```diff
@@ -81,13 +78,32 @@ where
 /// Naive version of the algorithm that given a full Network and a precomputed
 /// set W of random walks, iterates over each edge of the Network and computes
 /// the osrank.
-// pub fn osrank_naive(seed_set: SeedSet, network: &mut Network) -> Result<(), OsrankError> {
-//     // Phase1, rank the network and produce a NetworkView.
-//     let phase1 = random_walk(Some(seed_set), &NetworkView::from_network(network))?;
-//     // Phase2, compute the osrank only on the NetworkView
-//     let mut phase2 = random_walk(None, &phase1.network_view)?;
-//     rank_network(&phase2.walks, &mut phase2.network_view)
-// }
+pub fn osrank_naive<L, G>(
+    seed_set: Option<SeedSet>,
+    network: &mut G,
```

Comment:
This is fine for now, but perhaps something to think about for the "grand vision" -- It's conceivable that the Graph API won't give us a mutable reference to a graph, but just a read-only view, so we might have to perhaps return something that the Graph API can use to update the graph itself. This is all very hazy for now, but still work dwelling about it (cc @cloudhead ).

Reply:
Yes, I was thinking the same.

Reply:
I changed my mind a little on this - I think passing in a mutable graph makes sense - as it's much cheaper. What we could do though is isolate the osrank graph from the graph of balances etc.

Reply:
It's cheaper for sure, but I do wonder if that's a good idea. Perhaps we will have to experiment with both strategies and see which one yields the best tradeoff in terms of performance and security 😉